themes/default/content/blog/iac-recommended-practices-structuring-pulumi-projects/index.md

```diff
@@ -0,0 +1,105 @@
+
+Now that you have an idea of some of the factors that affect how users structure their Pulumi projects, what are the advantages and disadvantages of each approach?
+
+For single projects, the key advantage is _simplicity._ There's no denying that a single project is simpler and less complex than working with multiple projects. This advantage, though, is also the key disadvantage. A single project does not afford the flexibility to let different teams manage different aspects of the infrastructure, for example, nor does it offer the ability to split resources that don't have a shared lifecycle. You can't run `pulumi destroy` on only _some_ of the resources in a project.
+
+For multiple projects, the key advantage is _flexibility._ It's true that the added flexibility comes at the cost of some added complexity (a future post will explore one aspect of that added complexity in the form of StackReferences). However, you gain the ability to separate infrastructure resources according to team ownership/responsibility, relationship to other resources (i.e., grouping resources in a "shared infrastructure layer"), or any of the other dimensions mentioned above.
+
+Finally, there is the consideration of what happens in the event of human error, and limiting the so-called "blast radius." With a single project, _all_ the resources in that project are within the blast radius---all of them could potentially be affected in the event of human error. Using separate Pulumi projects can limit the blast radius. It's not something that should drive the decision all by itself, but it _is_ a factor to be considered.
+
+In some ways, the decision points for single vs. multiple projects are analogous to the decision points for monolith vs. microservice: independent scaling and flexibility versus simplicity.
+
+_In general, most customers end up using multiple Pulumi projects per use case._ As you've seen, though, there are a number of factors that influence that decision and will affect how many projects end up in use.
+
+## Moving forward with Zephyr
+
+Zephyr had a number of factors in play, not the least of which was the organizational change resulting from their growth and success. In the end, Zephyr decided to restructure their use of Pulumi for the online retail store into three separate projects:
+
+* An infrastructure project: This project handled the AWS VPC and related infrastructure resources.
+* A platform project: This project managed the Kubernetes cluster(s).
+* An application project: This project handled the actual online retail store itself.
+
+{{% notes %}}
+Note that today, Pulumi has no supported method to move resources from one project to another, so project restructuring will involve creating new infrastructure and migrating over to it.
+{{% /notes %}}
```

Comment:
What about importing an existing resource (with a `.get()`) and deleting it from the original stack?

Reply:
I tried a lot of different things, but in the end we decided the official statement we should make was this one. There _are_ ways to move resources, but they aren't necessarily supported ways.

Reply:
`pulumi import` is supported, and I think it's a reasonable alternative to this form of "create before destroy". 

Keep in mind for stateful infra like DBs, it's not necessarily practical to say "spin up a new DB and restore a backup". The DB in question could be ginormous.

Reply:
Fixed in e08be4c, let me know what you think of the updated wording